mrmustard/lab_dev/circuit_components.py

```diff
@@ -531,6 +539,33 @@ def to_fock(self, shape: int | Sequence[int] | None = None) -> CircuitComponent:
             del ret.manual_shape
         return ret
 
+    def to_bargmann(self) -> CircuitComponent:
+        r"""
+        Returns a new circuit component with the same attributes as this and a ``Bargmann`` representation.
+        .. code-block::
+
+            >>> from mrmustard.lab_dev import Dgate
+            >>> from mrmustard.physics.representations import Fock
+
+            >>> d = Dgate([1], x=0.1, y=0.1)
+            >>> d_fock = d.to_fock(shape=3)
+            >>> d_bargmann = d_fock.to_bargmann()
+
+
+            >>> assert d_bargmann.name == d.name
+            >>> assert d_bargmann.wires == d.wires
+            >>> assert isinstance(d_bargmann.representation, Bargmann)
+        """
+        # try:
+        #     bargmann = Bargmann(*self.bargmann_triple(batched=True))
+        #     bargmann._original_fock_data = self.representation.data
+        #     return self._from_attributes(bargmann, self.wires, self.name)
+        # except AttributeError:
+        A, b, _ = identity_Abc(len(self.wires))
+        bargmann = Bargmann(A, b, self.representation.data)
+        bargmann._original_fock_data = self.representation.data
+        return self._from_attributes(bargmann, self.wires, self.name)
```

Comment:
Should we have a 
```
if isinstance(self.representation, Bargmann):
     return self
```
somewhere here?

Reply:
I had that originally, but when everything can now be converted to bargmann I ended up having to use `to_fock()` way too much (so much it is no longer intuitive) to get mm to do what I want when contracting circuits. 

If we want the original bargmann data to be used, it should only be used in cases where the `c` in the triple is a scalar.

Reply:
I added this as a feature that we only use original bargmann data if c is a scalar.

Reply:
We ended up with a solution where we do not use the original bargmann data.